src/main/kotlin/racingcar/domain/RacingGame.kt

```diff
@@ -0,0 +1,26 @@
+package racingcar.domain
+
+class RacingGame(carCnt: Int, val tryCnt: Int) {
+    var cars = mutableListOf<Car>()
+
+    init {
+        repeat(carCnt) { cars.add(Car()) }
+    }
+
+    fun execute(method: () -> Int): List<CarHistory> {
+        val histories: MutableList<CarHistory> = mutableListOf()
+        cars.forEach {
+            val carHistory = it.move(method())
+            histories.add(carHistory)
+        }
+        return histories.toList()
```

Comment:
코틀린에서는 거의 가변리스트를 잘 안만들어 보는 습관을 만드는게 좋습니다.
그게 가능한것은 `map`등의 함수가 잘 제공되기 떄문입니다.

그래서 위의 코드는 다음과 같이 리팩터링이 가능합니다.

```suggestion
return cars.map { it.move(method()) }
```